src/diffusers/models/unet_2d_condition.py

```diff
@@ -247,16 +247,26 @@ def forward(
             res_samples = down_block_res_samples[-len(upsample_block.resnets) :]
             down_block_res_samples = down_block_res_samples[: -len(upsample_block.resnets)]
 
+            if upsample_block.upsamplers is not None:
+                upsample_size = down_block_res_samples[-1].shape[2:]
```

Comment:
This is to complicated here and makes the code very hard to understand / read. I think we should instead handle this over the config parameters (e.g. have upsample_size be saved inside the upsample_block after init)

Reply:
Hm, I don't think that would work - the correct upsample size varies depending on the width and height of the input `sample`. You can't save a fixed upsample size into the upsample block on its initialization and get correct results.

Reply:
I see! Thanks for clarifying